lib/swagger/blocks.rb

```diff
@@ -212,12 +212,15 @@ class Node
       attr_accessor :name
       attr_writer :version
 
-      def self.call(options = {}, &block)
+      def self.call(options = {}, props = {}, &block)
         # Create a new instance and evaluate the block into it.
         instance = new
         instance.name = options[:name] if options[:name]
         instance.version = options[:version]
-        instance.instance_eval(&block)
+        props.each_pair do |k,v|
```

Comment:
One big change: can you please rename `props` to `options` everywhere, and just pass the data through the options hash instead? It would make it more maintainable since we pass `version` options around already and we'd be able to use the same options hash in the future, instead of a new positional argument.